### PR TITLE
fix(k8s): error in cleanup-cluster-registry command and added test

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

One step of cleaning up the cluster registry was failing, which also prevented following steps to complete. Added a test against our cluster as well, so we should catch these issues from now on.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -374,10 +374,27 @@ jobs:
...
       - run:
           name: Cleanup
           command: CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM-docker kubectl delete --wait=false $(kubectl get ns -o name | grep testing-$CIRCLE_BUILD_NUM) || true
           when: always
+  cleanup-cluster-registry:
+    docker:
+      - image: gardendev/garden-gcloud:${CIRCLE_SHA1}
+        environment:
+          <<: *shared-env-config
+          GARDEN_TASK_CONCURRENCY_LIMIT: "10"
```
Comment (eysi09) [inline .circleci/config.yml:389]: Why do we need this here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -374,10 +374,27 @@ jobs:
...
           # overriding CIRCLE_BUILD_NUM to avoid conflict with other tests
           command: CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM-docker /garden/garden build --root examples/demo-project --env remote --logger-type basic
+      - run:
+          name: Run cluster cleanup
+          command: CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM-docker /garden/garden plugins kubernetes cleanup-cluster-registry --root examples/demo-project --env remote --logger-type basic
       - run:
           name: Cleanup
           command: CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM-docker kubectl delete --wait=false $(kubectl get ns -o name | grep testing-$CIRCLE_BUILD_NUM) || true
           when: always
+  cleanup-cluster-registry:
```
Comment (eysi09) [inline .circleci/config.yml:384]: Will this slow down our e2e tests in a meaningful way? Not the cleanup step itself but rather the next run on a "clean" registry. We of course don't delete images _in use_, so I'm assuming it won't really have an affect, but wanted to confirm.

File: .circleci/config.yml (hunk 1 above)
Comment (edvald) [reply .circleci/config.yml]: We don't, it was just in the copy-paste, now removed

File: .circleci/config.yml (hunk 2 above)
Comment (edvald) [reply .circleci/config.yml]: Not by a lot, but there are other issues with running the test like this. I've instead made it a part of the minikube tests, which is in a more closed environment and has no potential of conflicting with another test.

File: .circleci/config.yml [hunk 3]
```diff
@@ -374,6 +374,9 @@ jobs:
           name: Deploy demo-project with container
           # overriding CIRCLE_BUILD_NUM to avoid conflict with other tests
           command: CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM-docker /garden/garden build --root examples/demo-project --env remote --logger-type basic
+      - run:
```
Comment (eysi09) [inline .circleci/config.yml:377]: Doesn't this run against the CI cluster?

File: garden-service/static/kubernetes/system/docker-registry/garden.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@ name: docker-registry
 type: helm
 chart: stable/docker-registry
 releaseName: garden-docker-registry
-version: 1.8.3
+version: 1.9.2
```
Comment (eysi09) [inline garden-service/static/kubernetes/system/docker-registry/garden.yml:7]: Is this version already running in our cluster (to make sure the update doesn't break anything)?

**Comment by eysi09** [comment]
@edvald, had two outstanding questions :)

File: .circleci/config.yml (hunk 3 above)
Comment (edvald) [reply .circleci/config.yml:377]: Indeed. I'll fix.

File: garden-service/static/kubernetes/system/docker-registry/garden.yml (hunk 4 above)
Comment (edvald) [reply garden-service/static/kubernetes/system/docker-registry/garden.yml:7]: It is, yes.

**Comment by edvald** [comment]
@eysi09 should be good to go now

**Pull request merged**